### PR TITLE
Retroactive review of #271

### PR DESCRIPTION
**Pull request by negz**
Fixes https://github.com/crossplane/docs/issues/313

I didn't get a chance to review https://github.com/crossplane/docs/pull/271 very deeply. I spent some time this afternoon reviewing some of the new content. This PR contains a series of small tweaks. It may make sense to review at a commit-by-commit level. Some commit messages have context around why I'm proposing the changes.

### CODE REVIEW

File: content/v1.11/getting-started/introduction.md
Comment (jeanduplessis) [inline content/v1.11/getting-started/introduction.md:200]:
```suggestion
If a _Composite Resource Definition_ (`XRD`) allows a user to specify resource
settings, users apply them in a _Composite Resource_ (`XR`).
```

**suggestion (non-blocking)**: Since we're clarifying the difference between Composite Resource Definition and Composite Resource it would make sense to me to also add the Composite Resource abbreviation as means to reinforce it to the reader.

**Review by jeanduplessis — APPROVED** [review]
Approving with one non-blocking suggestion.

File: content/v1.11/getting-started/provider-aws.md
Comment (plumbis) [inline content/v1.11/getting-started/provider-aws.md:1177]:
```suggestion
The {{< hover label="xr" line="2">}}apiVersion{{< /hover >}} and {{< hover label="xr" line="3">}}kind{{</hover >}} are from the provider's CRDs.
```

File: content/v1.11/getting-started/provider-aws.md
Comment (plumbis) [inline content/v1.11/getting-started/provider-aws.md:1168]: What's the suffix to the generated resource? Will it look like the `kubectl get bucket` output below?

Comment (plumbis) [reply content/v1.11/getting-started/provider-aws.md:1168]: Also is there an easy/deterministic way to provide a `kubectl delete` command for the generated name?

**Review by plumbis — APPROVED** [review]
This is great, thank you for taking the time to do a thorough review. A few small notes but nothing blocking.

Can you update the pages in `content/master/` (you can just copy the whole file) as well?

Also may be worth a skim through AWS guides part 2 and 3 to ensure I'm sticking to the YAML styles you called out. I'll write an update to the writing guide to specifically mention this guideline.

File: content/v1.11/getting-started/introduction.md
Comment (negz) [reply content/v1.11/getting-started/introduction.md:407]: They can. I added this because I thought you had to include `namespace: default` in your manifest to create something in the default namespace. I just double checked and that's not the case. If you omit `metadata.namespace` kubectl will create your thing in the default namespace.

_Technically_ this is something `kubectl` is doing for you locally behind the scenes. At the Kubernetes API level you do need to specify a namespace (even if it's `default`). That said, it's probably safe to revert this change.

File: content/v1.11/getting-started/provider-aws.md
Comment (negz) [reply content/v1.11/getting-started/provider-aws.md:1168]: It will look quite similar - I think technically it's always a random 5 character suffix though. I think adding a label would be the best way to deterministically delete. I'll update.

**Comment by negz** [comment]
> Also may be worth a skim through AWS guides part 2 and 3 to ensure I'm sticking to the YAML styles you called out. I opened https://github.com/crossplane/docs/issues/330 to make sure this is covered in the style guide.

Good call. I did that and found quite a few extra things. I think it's worth another review pass before we merge.

> Can you update the pages in content/master/ (you can just copy the whole file) as well?

Yes, I'll do that once we're sure the v1.11 content is good to go. I don't want to push the issue right now, but FWIW this workflow does feel like a regression to me in comparison to using git branches.

File: content/v1.11/getting-started/provider-aws-part-2.md
Comment (negz) [inline content/v1.11/getting-started/provider-aws-part-2.md:475]: I removed this because I interpreted it the first 2-3 times as saying an XRD and a CRD are the same thing. I later realized it was saying an XRD is defined by a CRD. This is true, but I suspect a little mind-bending. I figured we could gloss over this here, especially given `kubectl explain` is the better/recommended way to explore a schema than `kubectl describe crd`.

File: content/v1.11/getting-started/provider-aws-part-2.md
Comment (negz) [inline content/v1.11/getting-started/provider-aws-part-2.md:519]: I'm not tied to this phrasing, but "allow a claim to use this XRD" doesn't seem accurate to me. It misses the context that the XRD is defining a type of claim into existence.

More broadly I think it's worth being aware of the different types of relationships we have and using more specific words than "uses". For example:

* XRD _defines_ XR
* XRD _defines_ XRC
* XRC _claims_ XR
* Composition _specifies_ MRs (maybe _templates_? I'm a little resistant to that word)
* XR _uses_ Composition (in this case "uses" does seem the most accurate to me)
* XR _composes_ MRs

File: content/v1.11/getting-started/provider-aws-part-2.md
Comment (negz) [inline content/v1.11/getting-started/provider-aws-part-2.md:371]: It may make sense to move (and the claim equivalent) this into introduction.md.

Comment (plumbis) [reply content/v1.11/getting-started/provider-aws-part-2.md:371]: I know that there is a practice of using X as a preface to defining the resource but I found it confusing as I was getting started. It is obscuring the thing we're talking about.

I have a custom API endpoint and I'm creating my definition of a database from it. That is straight forward. In the XRD and Composition concept chapters we can expand on this and explain why X.

File: content/v1.11/getting-started/provider-aws-part-2.md
Comment (plumbis) [reply content/v1.11/getting-started/provider-aws-part-2.md:519]: The point in the original text was to say "you can't have a claim if it's not defined in the XRD".  Maybe my mental model is wrong but I view an XRD as almost being an element with permissions. You can't access it via a claim unless the XRD _allows_ it.  

I'm open to changing it but "offers" doesn't feel like the right word here, that feels like a pub/sub thing, as if the XRD is producing an artifact (maybe it is, but it's not directly relevant to this part of the doc). I'm coming up blank on what to use an alternative. 
Tell this XRD to
 - allow a claim
 - allow the creation of claims 
 - create a claim type (this doesn't seem right) 
 - admit claims....something....

**Review by plumbis — APPROVED** [review]

**Comment by plumbis** [comment]
> FWIW this workflow does feel like a regression to me in comparison to using git branches.

For editing net-new content between `latest` and `master` I totally agree. For normal care and feeding the versions tend to drift as extensions and enhancements are applied to new versions making them slightly different from older versions. When it's time to fix a typo it's much easier to find and replace in a single repo/directory to fix it everywhere. My personal experience with backporting docs fixes in the old method was not fun.

A secondary question to raise is "should we have docs for master?" I'm not sure we provide much value to users by having master docs. They are drafts for the `v.NEXT` more than anything. Instead of having master docs I'd rather carve off a new `v.NEXT` folder and have that be the new draft space. Probably worth it to take this conversation outside of this PR.

File: content/v1.11/getting-started/provider-aws-part-2.md
Comment (negz) [reply content/v1.11/getting-started/provider-aws-part-2.md:519]: Let me try a dumb analogy - I'm at the beach making a sandcastle. The previous phrasing reads to me as equivalent to "Allow a sandcastle to use Nic to create it".

Where _offer_ comes from is the idea that:

* The platform team is defining a new API type.
* By default is not available to developers to claim. As an XR (only) it's private to the platform team.
* The platform team may choose to offer the API type to their end users. They do this by offering a claim.
* Offering a claim is done by specifying `spec.claimNames` on an XRD.

My preference would be to stick with the concept of "offering a claim". This is an established concept that permeated our previous documentation and shows up in Kubernetes events, errors, etc pertaining to the concept. e.g. https://github.com/search?q=org%3Acrossplane+offer&type=code

File: content/v1.11/getting-started/provider-aws-part-2.md
Comment (negz) [reply content/v1.11/getting-started/provider-aws-part-2.md:371]: Saying this convention is too confusing for the getting started guide but okay to introduce later doesn't sit well with me. If it's confusing it shouldn't exist. I'm also wary for the first experience most folks have with these concepts to differ from most other examples. All of Upbound's reference platforms use this pattern for example - https://marketplace.upbound.io/configurations/upbound/platform-ref-aws/v0.5.0.

Comment (negz) [reply content/v1.11/getting-started/provider-aws-part-2.md:371]: > I have a custom API endpoint and I'm creating my definition of a database from it. That is straight forward.

I'm not following this rationale at all, sorry.

Part of the reason we use the `X` convention is that, Crossplane concepts aside, the XR and the XRC represent the same thing. They're both API abstractions that represent "a database". "Database" and "CustomDatabase" doesn't make sense to me - one is not more custom than the other.

In an ideal world we might have both XR and XRC be of `kind: Database` with some other kind of differentiator, but Kubernetes makes that difficult. Early on we considered differentiating them by having the XR be `kind: Database` and the XRC be `kind: DatabaseClaim`. @bassam pointed out that that pattern is not ideal because the developers using this system probably just want to think about the claim as "my database", not "my database claim". The distinction between XRC and XR is more relevant to the platform engineers configuring Crossplane. So where we landed is that the claim gets `kind: Database` and the XR (which from the developer perspective is just "weird Crossplane machinery") gets the `kind: XDatabase` differentiator.

File: content/v1.11/getting-started/provider-aws-part-2.md
Comment (plumbis) [reply content/v1.11/getting-started/provider-aws-part-2.md:519]:
```suggestion
Tell this _XRD_ to offer developers a _claim_ by defining the _claim_ API endpoint under
the _XRD_ {{<hover label="XRDclaim" line="4">}}spec{{< /hover >}}.
```

The offer idea breakdown is super helpful. 

Let's just specific _who_ the claim is being offered to. It reads as if it's a Kubernets object offering something to another object. Can we just add "developers" to it's clear who the claim is being offered to? I'm sorry if this is too nit picky, I'm trying to write for someone like me who has little background knowledge on Crossplane and I'm trying to hammer out all of the assumptions or implied meanings.

File: content/v1.11/getting-started/provider-aws-part-2.md
Comment (plumbis) [reply content/v1.11/getting-started/provider-aws-part-2.md:371]: I'm not a fan of the convention in general but I'm the new guy. I've struggled with understanding who has the `database` and who has the `Xdatabase`. I'd prefer to using more unique names or indicating who is creating the resource, like `compositeDatabase`

I don't see a problem changing the terms since a guide like this is about introducing the minimum number of steps/topics and not about best practices. The goal here is for a reader to understand each `kind` and what is' responsible for. From there we can introduce more complex topics and best practices or standards.

Regarding the reference platforms, they are great if you know Crossplane. If you're new it's very hard to connect the dots. I don't see a problem keeping them in the reference platforms.

**Comment by negz** [comment]
@plumbis Do you feel okay merging this PR and following up on our open threads separately? I'm assuming yes because I see you re-approved the PR but I want to make sure.

**Comment by plumbis** [comment]
Yeah, no objections.

**Pull request merged**
